cli/azd/pkg/config/config.go

```diff
@@ -31,3 +33,98 @@ func GetUserConfigDir() (string, error) {
 
 	return configDirPath, err
 }
+
+type Config struct {
+	DefaultSubscription *Subscription `json:"defaultSubscription"`
+	DefaultLocation     *Location     `json:"defaultLocation"`
+}
+
+type Subscription struct {
+	Id       string `json:"id"`
+	Name     string `json:"name"`
+	TenantId string `json:"tenantId"`
+}
+
+type Location struct {
+	Name        string `json:"name"`
+	DisplayName string `json:"displayName"`
+}
+
+// Saves the users configuration to their local azd user folder
+func (c *Config) Save() error {
+	configJson, err := json.Marshal(*c)
+	if err != nil {
+		return fmt.Errorf("failed marshalling config JSON: %w", err)
+	}
+
+	configPath, err := getConfigFilePath()
+	if err != nil {
+		return fmt.Errorf("failed locating config dir")
+	}
+
+	err = os.WriteFile(configPath, configJson, osutil.PermissionFile)
+	if err != nil {
+		return fmt.Errorf("failed saving configuration JSON: %w", err)
+	}
+
+	return nil
+}
+
+// Loads azd configuration from the users configuration dir
+func Load() (*Config, error) {
+	configPath, err := getConfigFilePath()
+	if err != nil {
+		return nil, fmt.Errorf("failed locating config dir")
+	}
+
+	bytes, err := os.ReadFile(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading project file: %w", err)
+	}
+
+	return Parse(bytes)
+}
+
+// Parses azd configuration JSON and returns a Config instance
+func Parse(configJson []byte) (*Config, error) {
+	var config Config
+	err := json.Unmarshal(configJson, &config)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshalling configuration JSON: %w", err)
+	}
+
+	return &config, nil
+}
+
+func getConfigFilePath() (string, error) {
+	configPath, err := GetUserConfigDir()
+	if err != nil {
+		return "", fmt.Errorf("failed locating config dir")
+	}
+
+	return filepath.Join(configPath, "config.json"), nil
+}
+
+type contextKey string
+
+const configContextKey contextKey = "config"
+
+// Gets the AZD config from current context
+// If it does not exist will return a new empty AZD config
+func GetConfig(ctx context.Context) *Config {
+	config, ok := ctx.Value(configContextKey).(*Config)
+	if !ok {
+		loadedConfig, err := Load()
+		if err != nil {
+			loadedConfig = &Config{}
+		}
```

Comment:
Catching all types of `err` and defaulting to empty config will lead to unintended behaviors like:
1. Filesystem read fails
2. Default back to empty config
3. On save, we delete existing data and rewrite it with an empty config

Reply:
Would prefer not introducing config until we need to (and having a more targeted change that can more holistically address the needs of having a config)

Reply:
Removed all our commands for saving config at the moment.  We can see if/how/when we add config.